src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -697,6 +766,9 @@ void ImageReplayer<I>::on_stop_local_image_close_finish(int r)
   m_replay_status_formatter = nullptr;
 
   m_remote_journaler->stop_replay();
+  if (m_stopping_for_resync) {
+    m_remote_journaler->unregister_client();
```

Comment:
I that case, when the rbd-mirror recovers it will check in the bootstrap that it has to resync and will finish the work. 
And now that I'm thinking, I forgot to update the `request_resync` flag from the journal metadata, after a successful resync.